### PR TITLE
VerticalStackLayout and HorizontalStackLayout

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change ###

Okay, this is some really, really basic layout stuff so that we can start getting feedback/bugs and get a stake in the ground, layout-wise.

This includes VerticalStackLayout and HorizontalStackLayout; they work pretty much like you'd expect. Each one also includes a Spacing property to provide equal inter-item spacing. They are deliberately separated to make the implementations as straightforward as possible. There are probably lots of minor performance tweaks to be had, but we'll worry about those once we've fully established the behavior and properties.  

Eventually the Forms implementation may (probably will) include a StackLayout implementation with a bindable Orientation property which will internally use these classes. 

IFrameworkElement adds Width and Height - these are replacements for WidthRequest and HeightRequest, and function as much stronger suggestions about the eventual size of the control. This does not include Minimum/Maximum Width/Height yet.

### CODE REVIEW

File: src/Platform.Handlers/src/Xamarin.Platform.Handlers/Platform/Android/LayoutViewGroup.Android.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+			{
+				return;
+			}
+
+			var deviceIndependentLeft = Context.FromPixels(l);
+			var deviceIndependentTop = Context.FromPixels(t);
+			var deviceIndependentRight = Context.FromPixels(r);
+			var deviceIndependentBottom = Context.FromPixels(b);
+
+			var destination = Rectangle.FromLTRB(deviceIndependentLeft, deviceIndependentTop,
```
Comment (AmrAlSayed0) [inline src/Platform.Handlers/src/Xamarin.Platform.Handlers/Platform/Android/LayoutViewGroup.Android.cs:69]: Account for RTL? #3066 🤔

Comment (hartez) [reply src/Platform.Handlers/src/Xamarin.Platform.Handlers/Platform/Android/LayoutViewGroup.Android.cs:69]: When we implement margins, we will need to account for it.

File: src/Platform.Handlers/src/Xamarin.Platform.Handlers/Platform/Android/LayoutViewGroup.Android.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+namespace Xamarin.Platform.Handlers
+{
+	public class LayoutViewGroup : ViewGroup
+	{
+		public LayoutViewGroup(Context context) : base(context)
+		{
+		}
+
+		public LayoutViewGroup(IntPtr javaReference, JniHandleOwnership transfer) : base(javaReference, transfer)
```
Comment (PureWeen) [inline src/Platform.Handlers/src/Xamarin.Platform.Handlers/Platform/Android/LayoutViewGroup.Android.cs:17]: Are these here intentionally for cases where the layout will be inflated from an XML file?

**Review by PureWeen — APPROVED** [review]

File: src/Platform.Handlers/src/Xamarin.Platform.Handlers/Core/IStackLayout.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Xamarin.Platform
+{
+	public interface IStackLayout : ILayout
```
Comment (rmarinho) [inline src/Platform.Handlers/src/Xamarin.Platform.Handlers/Core/IStackLayout.cs:3]: Grid also has spacing should be this ISpacingLayout

**Comment by KPixel** [comment]
What do you think of the idea of splitting the `Layout` class into two parts: `ItemsView` which hosts the children elements, and `Layout` classes which contain the "layout algorithms"?

The immediate benefit is the ability to change layouts without having to swap the control. This can be very powerful when writing responsive apps. For example, an app can use a stack layout when running on a small screen/window, and switch to a wrap layout when the screen/window is larger. It also makes it easier to plug in a different layout by just writing the (cross platform) algorithm for it.

For backward compatibility reasons, the actual implementation would be to keep the current Layout with both parts, and introduce these new concepts behind the scene. Then, old apps will continue to work as expected, and new apps can directly tap into these new concepts.

**Review by Clancey — APPROVED** [review]

**Pull request merged**